### ISSUE TITLE
why custom Instance Segmentation Train Does not converge

### ISSUE THREAD

**Issue by fanweiya** [ayoolaolafenwa/PixelLib#22]
### this is my datasets.[googledrive](https://drive.google.com/file/d/1tLRlX2vOKzbCg-IlcL3afGL6LgF9S7Uc/view?usp=sharing)

**Comment by ayoolaolafenwa** [comment]
I don't understand what you meant by changing the learning rate. I did not provide the option to change learning rate in PixelLib. It is only possible to change the target minimum and maximum image shape. Also I cannot access your google drive to check the dataset you used. 
I have requested to access your drive with the email olafenwaayoola@gmail.com.

**Comment by fanweiya** [comment]
I midified config file.Also, I  send you my dataset by email.

**Comment by ayoolaolafenwa** [comment]
> 
> 
> I midified config file.Also, I send you my dataset by email.

Okay I will check it. By the way next time do not attempt to modify any parameter.

**Comment by fanweiya** [comment]
okay,I will be.

**Comment by ayoolaolafenwa** [comment]
I have checked the dataset. The problem is this you do not follow the requirements for training images. I realized that your dataset is made up of two classes (live and dead cells) and there are only 21 images for training and 6 images for testing. 
If you read my [tutorial](https://github.com/ayoolaolafenwa/PixelLib/blob/master/Tutorials/custom_train.md) on custom training  I specified that the minimum images required for training is 300  for each class in a dataset. The nature dataset I used as a  sample in this tutorial has two classes, 300 images for each class for training and 100 for each class for testing.  The solution is this, you should increase the number of images for each class in your dataset to at least 300 and if possible it should be more than that. You should also train for a longer range range of epochs like 100-300 epochs to achieve the lowest validation loss possible.

**Comment by fanweiya** [comment]
> I have checked the dataset. The problem is this you do not follow the requirements for training images. I realized that your dataset is made up of two classes (live and dead cells) and there are only 21 images for training and 6 images for testing. 
> 
> If you read my [tutorial](https://github.com/ayoolaolafenwa/PixelLib/blob/master/Tutorials/custom_train.md) on custom training  I specified that the minimum images required for training is 300  for each class in a dataset. The nature dataset I used as a  sample in this tutorial has two classes, 300 images for each class for training and 100 for each class for testing.  The solution is this, you should increase the number of images for each class in your dataset to at least 300 and if possible it should be more than that. You should also train for a longer range of epochs like 100-300 epochs to achieve the lowest validation loss possible.

Although my datasets has few pictures, there are many instances of each picture. I think this is not a problem that can be solved by adding data. And I also trained a lot of epochs, but the loss on the validation set has been rising until infinity.

**Comment by ayoolaolafenwa** [comment]
> > I have checked the dataset. The problem is this you do not follow the requirements for training images. I realized that your dataset is made up of two classes (live and dead cells) and there are only 21 images for training and 6 images for testing.
> > If you read my [tutorial](https://github.com/ayoolaolafenwa/PixelLib/blob/master/Tutorials/custom_train.md) on custom training  I specified that the minimum images required for training is 300  for each class in a dataset. The nature dataset I used as a  sample in this tutorial has two classes, 300 images for each class for training and 100 for each class for testing.  The solution is this, you should increase the number of images for each class in your dataset to at least 300 and if possible it should be more than that. You should also train for a longer range of epochs like 100-300 epochs to achieve the lowest validation loss possible.
> 
> Although my datasets has few pictures, there are many instances of each picture. I think this is not a problem that can be solved by adding data. And I also trained a lot of epochs, but the loss on the validation set has been rising until infinity.

I have mentioned the requirements  for a successful training and there is nothing else I can advise you to do.

I proposed  at least 300 images for each class, you used like 10 images for each class in your dataset and expecting good results. Unfortunately it does not work like this.

**Comment by fanweiya** [comment]
Thanks
